### ISSUE TITLE
Nested scopes

### ISSUE THREAD

**Issue by AlexParamonov** [zipmark/rspec_api_documentation#198]
Hello.

It would be great to do 
```
parameter :links, "Linked objects", scope: :articles
parameter :author, "Author id", scope: :links
```

http://jsonapi.org/format/#crud-updating-relationships

**Comment by oestrich** [comment]
This could be cool. When I get time I will look into this.

**Comment by juniorjp** [comment]
I was just looking into this, any workaround?

**Comment by xdmx** [comment]
:+1: on this
maybe something like this?

```
parameter :links, "Linked objects", scope: :articles
parameter :author, "Author id", scope: [:articles, :links]
```

**Comment by kevintraver** [comment]
Perhaps something like this:
```
parameter :author, "Author id", scope: { :articles => :links }
```

and then do some sort of merge into the existing hash

```ruby
def set_param(hash, param)
  key = param[:name]
  return hash if !respond_to?(key) || in_path?(key)

  scope = param[:scope]
  if scope
    if scope.is_a?(Hash)
      # somehow merge into hash
    else
      hash[scope.to_s] ||= {}
      hash[scope.to_s][key] = send(key)
    end
  else
    hash[key] = send(key)
  end

  hash
end
```

https://github.com/zipmark/rspec_api_documentation/blob/3c2f23d7442defc326c6b8674ec614e9260bc7d8/lib/rspec_api_documentation/dsl/endpoint.rb#L149-L158

**Comment by oestrich** [comment]
Closed with #221
